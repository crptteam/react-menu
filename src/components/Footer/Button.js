import React from 'react';
import PropTypes from 'prop-types';
import ButtonIcon from '../../svg/buttonIcon.svg';
import ButtonWrapper from '../../styled/ButtonWrapper';

const propTypes = {
  onClick: PropTypes.func,
};

const defaultProps = {
  onClick: () => {},
};

const Button = ({
  onClick,
}) => (
  <ButtonWrapper onClick={onClick}>
    <ButtonIcon />
  </ButtonWrapper>
);

Button.displayName = 'Button';
Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
