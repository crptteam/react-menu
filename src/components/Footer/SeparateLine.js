import React from 'react';
import SeparateLineWrapper from '../../styled/SeparateLineWrapper';


const SeparateLine = () => (<SeparateLineWrapper />);
SeparateLine.displayName = 'SeparateLine';

export default SeparateLine;
