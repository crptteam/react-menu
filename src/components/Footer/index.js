import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import FooterWrapper from '../../styled/FooterWrapper';
import Button from './Button';
import SeparateLine from './SeparateLine';


export default class Footer extends PureComponent {
  displayName = 'Footer';

  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    content: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
  };

  render() {
    const {
      isOpen,
      onClick,
      content: FooterContent,
    } = this.props;

    return (
      <FooterWrapper>
        <Button onClick={onClick} />
        <SeparateLine />
        <FooterContent isOpen={isOpen} />
      </FooterWrapper>
    );
  }
}
