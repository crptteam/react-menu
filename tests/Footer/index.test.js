import React from 'react';
import { shallow } from 'enzyme';

import Footer from '../../src/components/Footer';

describe('Footer', () => {
  it('should renders without problem', () => {
    const content = jest.fn();
    const onClick = jest.fn();
    const isShowButton = jest.fn();
    const component = shallow(<Footer isOpen content={content} onClick={onClick} isShowButton={isShowButton} />);
    expect(component).toMatchSnapshot();
  });

  it('calling content render with isOpen', () => {
    const content = jest.fn();
    const onClick = jest.fn();
    const isShowButton = jest.fn();
    const component = shallow(<Footer isOpen content={content} onClick={onClick} isShowButton={isShowButton} />);
    expect(
      component
        .find('mockConstructor')
        .at(0)
        .prop('isOpen'),
    ).toEqual(true);

    component.setProps({ isOpen: false, content, onClick });
    expect(
      component
        .find('mockConstructor')
        .at(0)
        .prop('isOpen'),
    ).toEqual(false);
  });

  it('calling callback on click to Button', () => {
    const content = jest.fn();
    const onClick = jest.fn();
    const isShowButton = jest.fn();
    const component = shallow(<Footer isOpen content={content} onClick={onClick} isShowButton={isShowButton} />);
    component.find('Button').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
