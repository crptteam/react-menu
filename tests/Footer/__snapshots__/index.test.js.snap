// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should renders without problem 1`] = `
<WithTheme(FooterWrapper)>
  <Button
    onClick={[MockFunction]}
  />
  <SeparateLine />
  <mockConstructor
    isFull={true}
  />
</WithTheme(FooterWrapper)>
`;
